app/src/main/kotlin/at/bitfire/davdroid/sync/Syncer.kt

```diff
@@ -96,15 +96,20 @@ abstract class Syncer<CollectionType: LocalCollection<*>>(
 
         // Find collections in database and provider which should be synced (are sync-enabled)
         val dbCollections = getSyncEnabledCollections()
-        val localCollections = getLocalCollections(provider).toMutableList()
+        var localCollections = getLocalCollections(provider).toMutableList()
 
         // Update/delete local collections and determine new (unknown) remote collections
-        val newDbCollections = updateCollections(localCollections, dbCollections)
+        val (newDbCollections, deletedLocalCollections) = updateCollections(localCollections, dbCollections)
 
         // Create new local collections for newly found remote collections
         val newProviderCollections = createLocalCollections(provider, newDbCollections)
         localCollections.addAll(newProviderCollections) // Add the newly created collections
 
+        // Don't sync deleted collections
+        deletedLocalCollections.forEach { deletedLocalCollection ->
+            localCollections.remove(deletedLocalCollection)
```

Comment:
In order to have the actual deletion and the deletion from to-be-synced-list at the same place, this could happen outside of `updateCollections` where we delete from the list too.